controllers/profiles/object_creators.go

```diff
@@ -202,3 +208,54 @@ func ensureWorkflowSpecConfigMapMutator(workflow *operatorapi.KogitoServerlessWo
 		}
 	}
 }
+
+// workflowPropsConfigMapCreator creates a ConfigMap to hold the external application properties
+func workflowPropsConfigMapCreator(workflow *operatorapi.KogitoServerlessWorkflow) (client.Object, error) {
+	return &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      getWorkflowPropertiesConfigMapName(workflow),
+			Namespace: workflow.Namespace,
+			Labels:    labels(workflow),
+		},
+		// we could use utils.NewJavaProperties, but this way is faster
+		Data: map[string]string{
+			quarkusConfigFileName: "# Immutable Workflow Properties \n" +
+				"quarkus.http.port=" + strconv.Itoa(defaultHTTPWorkflowPort) + "\n" +
+				"quarkus.http.host=0.0.0.0\n",
+		},
+	}, nil
+}
+
+func getWorkflowPropertiesConfigMapName(workflow *operatorapi.KogitoServerlessWorkflow) string {
+	return workflow.Name + workflowConfigMapNameSuffix
+}
+
+func ensureWorkflowPropertiesConfigMapMutator(workflow *operatorapi.KogitoServerlessWorkflow) mutateVisitor {
+	return func(object client.Object) controllerutil.MutateFn {
+		return func() error {
+			if kubeutil.IsObjectNew(object) {
+				return nil
+			}
+			original, err := workflowPropsConfigMapCreator(workflow)
+			if err != nil {
+				return err
+			}
+			cm := object.(*corev1.ConfigMap)
```

Comment:
We are 100% that the object is a ConfigMap but I'll double check the type assertion result in order to avoid a panic.

Reply:
It's better to panic rather than silently fail in this case (or just a log in the console) since a conversion error here is an implementation problem.